llm_creator.py

```
@@ -0,0 +1,128 @@
+            
+        Returns:
+            An instance of a BaseLLM implementation
+            
+        Raises:
+            ValueError: If an unsupported provider is specified
+        """
+        provider = provider.lower()
+        
+        # Create OpenAI LLM
+        if provider == "openai":
+            return OpenAILLM(
+                model_name=model_name,
+                api_key=api_key or os.getenv("OPENAI_API_KEY"),
+                base_url=base_url,
+                **kwargs
+            )
+        
+        # Create local LLM (Ollama, etc.)
+        elif provider == "local":
+            return LocalLLM(
+                model_name=model_name,
+                base_url=base_url or "http://localhost:11434/v1",
+                api_key=api_key,
+                **kwargs
+            )
+        
+        # Other providers could be added here
+        
+        else:
+            raise ValueError(f"Unsupported LLM provider: {provider}")
+    
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> BaseLLM:
+        """Create an LLM from a configuration dictionary.
+        
+        Args:
+            config: Dictionary containing LLM configuration
+                Required keys:
+                - provider: LLM provider name
+                - model_name: Name of the model
+                Optional keys:
+                - api_key: API key
+                - base_url: Custom endpoint
+                - Any other provider-specific parameters
+                
+        Returns:
+            An instance of a BaseLLM implementation
+        """
+        provider = config.pop("provider")
+        model_name = config.pop("model_name")
+        
+        # Extract common parameters
+        api_key = config.pop("api_key", None)
+        base_url = config.pop("base_url", None)
+        
+        # Pass remaining config as kwargs
+        return cls.create_llm(
+            provider=provider,
```

In from_core_settings, handle cases where core_settings.LLM_MODEL_CONFIG is missing or None. Validate presence of required attributes (MODEL_NAME, OPENAI_MODEL, BASE_URL). Raise clear configuration errors if settings are incomplete to avoid silent failures.